tests/attacks/test_knockoff_nets.py

```diff
@@ -0,0 +1,434 @@
+
+import logging
+import unittest
+
+import tensorflow as tf
+import numpy as np
+import keras.backend as k
+import keras
+from keras.models import Sequential
+from keras.layers import Dense, Flatten, Conv2D, MaxPooling2D
+
+from art.attacks.extraction.knockoff_nets import KnockoffNets
+from art.classifiers import TensorFlowClassifier
+from art.classifiers import KerasClassifier
+from art.classifiers import PyTorchClassifier
+from art.utils import load_dataset, master_seed
+from tests.utils_test import get_classifier_tf
+from tests.utils_test import get_classifier_kr
+from tests.utils_test import get_classifier_pt
+from tests.utils_test import get_iris_classifier_tf
+from tests.utils_test import get_iris_classifier_kr
+from tests.utils_test import get_iris_classifier_pt
+from art.config import ART_NUMPY_DTYPE
+
+logger = logging.getLogger(__name__)
+
+try:
+    # Conditional import of `torch` to avoid segmentation fault errors this framework generates at import
+    import torch
+    import torch.nn as nn
+    import torch.optim as optim
+except ImportError:
+    logger.info('Could not import PyTorch in utilities.')
+
+if tf.__version__[0] == '2':
+    # pylint: disable=E0401
+    import tensorflow.compat.v1 as tf
+    tf.disable_eager_execution()
+
+
+BATCH_SIZE = 100
+NB_TRAIN = 1000
+NB_EPOCHS = 10
+NB_STOLEN = 1000
+
+
+class TestKnockoffNets(unittest.TestCase):
+    """
+    A unittest class for testing the KnockoffNets attack.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        (x_train, y_train), (_, _), _, _ = load_dataset('mnist')
+
+        cls.x_train = x_train[:NB_TRAIN].astype(ART_NUMPY_DTYPE)
+        cls.y_train = y_train[:NB_TRAIN].astype(ART_NUMPY_DTYPE)
+
+    def setUp(self):
+        master_seed(1234)
+
+    def test_tfclassifier(self):
+        """
+        First test with the TensorFlowClassifier.
+        :return:
+        """
+        # Build TensorFlowClassifiers
+        victim_tfc, sess = get_classifier_tf()
+
+        # Define input and output placeholders
+        input_ph = tf.placeholder(tf.float32, shape=[None, 28, 28, 1])
+        output_ph = tf.placeholder(tf.int32, shape=[None, 10])
+
+        # Define the tensorflow graph
+        conv = tf.layers.conv2d(input_ph, 1, 7, activation=tf.nn.relu)
+        conv = tf.layers.max_pooling2d(conv, 4, 4)
+        flattened = tf.layers.flatten(conv)
+
+        # Logits layer
+        logits = tf.layers.dense(flattened, 10)
+
+        # Train operator
+        loss = tf.reduce_mean(tf.losses.softmax_cross_entropy(logits=logits, onehot_labels=output_ph))
+        optimizer = tf.train.AdamOptimizer(learning_rate=0.001)
+        train = optimizer.minimize(loss)
+
+        # TensorFlow session and initialization
+        sess.run(tf.global_variables_initializer())
+
+        # Create the classifier
+        thieved_tfc = TensorFlowClassifier(clip_values=(0, 1), input_ph=input_ph, output=logits, labels_ph=output_ph,
```

Comment:
Would it be possible to create `thieved_tfc` with `get_classifier_tf`?

Reply:
Solved.